Sprint 8

kazakovdmitriy opened the pull request:
Выполнены задания по 8 спринту

Commit:
Add splash screen

Commit:
Сделан экран загрузки, выполнен главный экран с таблицей, добавлен градиент

So1ovyov commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,84 @@
+        cell.cellImageOutlet.image = image
+        cell.dateLabelOutlet.text = currentDate
+        cell.likeButtonOutlet.setImage(likeImage, for: .normal)
+        cell.dateBackgroundOutlet.setGradientBackground(startColor: .clear, endColor: UIColor.blackYp, opacity: 0.2)
```
При реализации градиента слой добавляется при каждом вызове `configCell`, который в свою очередь вызывается при переиспользовании ячейки.

- Это плохо для перформанса. Добавленные слои могут скапливаться до бесконечности, таблица может начать тормозить
- Это дает такой некрасивый эффект налажения градиента до практически черного прямоугольника. 

Чтобы исправить нужно:

- либо создавать слой один раз при инициализации ячейки (не конфигурации)
- либо удалять старый слой в конфигурации. Перед этим его сохранять для ячейки. 
Можно удалять слой в специальном методе `prepareForReuse`, который вызывается перед началом переиспользования. 

Если возникнут вопросы, смело задавай их Наставнику!

So1ovyov commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,84 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.rowHeight = 200
```
В этой строке нет необходимости и её можно удалить, так как высоту ячейки ты рассчитываешь в методе `heightForRowAt`

So1ovyov commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,84 @@
+        let imageName = photosName[indexPath.row]
+        guard let image = UIImage(named: imageName) else { return }
+        
+        let currentDate = dateFormatter.string(from: .now)
```
Так как ты используешь эту константу 1 раз ниже, предлагаю избавиться от этой строки и написать сразу 
```
cell.dateLabelOutlet.text = dateFormatter.string(from: .now)
```

So1ovyov commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,84 @@
+}
+
+extension ImagesListViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
```
Также предлагаю избавиться от этого метода, так как логики он не содержит.

Commit:
Исправлен версия ios на 13, основной таргет только iphone, ориентация портретная
Исправил отступы между ячейками
Добавил скругление градиенту, так же слой градиента накладывается только при инициализации ячейки

kazakovdmitriy commented:
Исправил критические замечания, также внес изменения по замечаниям из ревью

Commit:
Добавление градиента вынесено в метод layoutSybviews, чтобы оно появлялось сразу при инициализации view
